### DISCUSSION TITLE
Amount of flash memory required by OpenThread for dataset

### DISCUSSION THREAD

**Discussion by ValNyrov** [openthread/openthread#8076] [Q&A] [answered]
Hi all!

I'm new and just learning. Tell me, what is the exact amount of flash memory required by OpenThread to store the configuration (for Operational Datasets, maybe for something else)? Where can I read about it?

BR,
Valeriy

**Comment by jwhui** [comment, marked as answer]
The Thread Conformance Specification states that the maximum size for Operational Datasets is 254 bytes. In OpenThread, this is reflected at:
https://github.com/openthread/openthread/blob/0f10480ed616f9d23a73f65579de838d4efa5d5e/include/openthread/dataset.h#L254

OpenThread stores non-volatile information using the [`otPlatSettings*()`](https://openthread.io/reference/group/plat-settings) interface. The `OT_SETTINGS_KEY_*` enum identifies the set of items that OpenThread stores in non-volatile. The amount of non-volatile storage needed will depend on various build configurations (e.g. child table size, use of SRP client, etc.).

**Reply by ValNyrov** [reply]
Thanks for the help!
What is the maximum size of the child/router ID table?
BR,
Valeriy

**Reply by jwhui** [reply]
Thread Protocol supports up to 511 children per router. OpenThread has a build-time configuration:
https://github.com/openthread/openthread/blob/cba1bebf1f1ef3531bfe5397c75c3c6e045218b6/src/core/config/mle.h#L58

Thread Protocol has 63 valid router IDs, but only 32 may be active at a given time.

**Reply by ValNyrov** [reply]
Thanks!
